presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -549,8 +541,10 @@ public ColumnMetadata getColumnMetadata(Session session, TableHandle tableHandle
                             entry.getKey().getTableName());
 
                     ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
-                    for (ViewColumn column : deserializeView(entry.getValue().getViewData()).getColumns()) {
-                        columns.add(new ColumnMetadata(column.getName(), column.getType()));
+                    for (ViewColumn column : entry.getValue().getColumns()) {
+                        Type type = getType(column.getType());
+                        checkCondition(type != null, INVALID_VIEW, "Unknown type '%s' for column '%s' in view: %s", column.getType(), column.getName(), tableName);
```

Comment:
I'm pretty sure that `getType` throws an `IllegalArgumentException` on an unknown type:

https://github.com/prestosql/presto/blob/master/presto-main/src/main/java/io/prestosql/type/TypeRegistry.java#L208

Reply:
That's not what the Javadoc says: https://github.com/prestosql/presto/blob/master/presto-spi/src/main/java/io/prestosql/spi/type/TypeManager.java#L26

Reply:
Looks like that changed somewhat recently, but the Javadoc and some call sites were missed.